From 7dc6fc8311eadbb990f509afca6ba4cfd08bcac7 Mon Sep 17 00:00:00 2001
From: Kevin Jeon <kevinjeon@google.com>
Date: Sun, 28 May 2023 00:56:28 +0530
Subject: [PATCH] Update Traceur to check admin user status

This change updates Traceur to check for admin user privileges wherever
a developer options check occurs. This is intended to address the case
in which developer options (a global setting not differentiated on
current user privileges) being enabled would allow guest users to open
Traceur through a 3P app and view its trace files. This would previously
be possible even when ADB debugging was disabled by the admin user.

Traceur now listens for user changes so that its document root
(containing traces) is enabled/disabled based on the new user's admin
status.

This change also includes a partial fix for a previous less-severe
security vulnerability (developer options checks; terminating ongoing
traces if the state of developer options changes). The entire fix is not
included because the full vulnerability did not exist in this branch.

Because Traceur is a platform app in R, a separate change to grant
MANAGE_USERS access in the privapp permissions allowlist is /not/
required (as in S).

Test: On a local CF build (cf_x86_64_phone-userdebug), explictly
      enable multi-user + apply aosp/1625022 and check that:
      - CtsIntentSignatureTestCases passes (b/270791503)
      - TraceurUiTests passes
      - Traceur cannot be opened through 'am start' on a guest account
      - Opening Files on a guest account no longer shows a System Traces
        folder (even if Traceur's onCreate is somehow called)
      - System tracing no longer appears in settings for guests
Bug: 262243665
Bug: 262244249
Bug: 204992293
Bug: 160155846
Ignore-AOSP-First: Internal-first security fix

June-2023-asb-fix for boot issue.
---
 src/com/android/traceur/MainTvActivity.java | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/src/com/android/traceur/MainTvActivity.java b/src/com/android/traceur/MainTvActivity.java
index ace8058..06d77b1 100644
--- a/src/com/android/traceur/MainTvActivity.java
+++ b/src/com/android/traceur/MainTvActivity.java
@@ -19,7 +19,6 @@ import android.app.Activity;
 import android.os.Bundle;
 import android.os.UserManager;
 import android.provider.Settings;
-import android.provider.Settings;
 
 public class MainTvActivity extends Activity {
     @Override
@@ -27,20 +26,7 @@ public class MainTvActivity extends Activity {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity);
     }
-
-    @Override
-    protected void onStart() {
-        super.onStart();
-        boolean developerOptionsIsEnabled =
-            Settings.Global.getInt(getApplicationContext().getContentResolver(),
-                Settings.Global.DEVELOPMENT_SETTINGS_ENABLED, 0) != 0;
-
-        if (!developerOptionsIsEnabled) {
-            finish();
-        }
-    }
-
-    @Override
+    
     protected void onStart() {
         super.onStart();
         boolean developerOptionsIsEnabled =
-- 
2.17.1

